packages/workshop-app/app/components/edit-readme.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import { LaunchEditor } from '~/routes/launch-editor.tsx'
+
+export function EditReadme({
+	name,
+	stepPath,
+}: {
+	name: string
+	stepPath: string
+}) {
+	const [altDown, setAltDown] = React.useState(false)
+
+	React.useEffect(() => {
+		if (ENV.KCDSHOP_DEPLOYED) return
+		const set = (e: KeyboardEvent) => setAltDown(e.altKey)
+		document.addEventListener('keydown', set)
+		document.addEventListener('keyup', set)
+		return () => {
+			document.removeEventListener('keyup', set)
+			document.removeEventListener('keydown', set)
+		}
+	}, [])
+
+	// prevent Alt+click from trying to save the file when KCDSHOP_DEPLOYED is true
```

Comment:
If alt is down then it will render the launch button. If it's not this won't be a problem. I think we can render the link.

Reply:
When ` ENV.KCDSHOP_DEPLOYED` we don't want to open the the link in local editor event with alt. notice that i skip the useEffect when  ENV.KCDSHOP_DEPLOYED is true. if we use <Link> or <a> when alt is down the browser open dialog to save the file.

my solution to this is to use button to open new tab in Github

Reply:
if you still want to use `LaunchEditor` when `ENV.KCDSHOP_DEPLOYED` is true, we will have to export LaunchEditorImpl and call it from edit-readme instead of calling `LaunchEditor`

Reply:
Oh, I see. Yeah, I'd just really prefer that if something *goes somewhere* it should use a link. When deployed the `alt` feature shouldn't work anyway since both will go to github. We could probably just change the condition below to be `altDown && !ENV.KCDSHOP_DEPLOYED` and then we can using a link instead of a button.

Reply:
but when you use Link with alt down the browser try to save the file instead open a tab

Reply:
Right, but we won't use that feature in the deployed app anyway

Reply:
look like we have a misunderstanding.
that is how I understand it

 on non deployed app:
  1a. with Alt modifier down - open file in local editor 
  1b. without modifier - open file in Github 

 on deployed app:
  2. **always** - open file in Github
> but we won't use that feature in the deployed app anyway

unless you prefer to hide the edit button in this case 

we don't have problem with cases 1a and 1b.
1a - use `LaunchEditor` and 1b - can use <Link> since we check the Alt was not down

for case 2 - we want to open the file in Github but we don't have control on the modifier use hold down. if user hold down Alt - the browser will try to save the file.

try the link `Edit this page on GitHub` from https://tailwindcss.com/docs/will-change and hold `Alt`

Reply:
I'm saying in a deployed app I would never hold down alt. Why would I? If I hold alt when clicking a link I expect it to download a file. So I wouldn't do it. Holding down alt in the deployed case doesn't change anything so I wouldn't hold down alt in the deployed app so we don't need to worry about it.

Reply:
👍